controllers/post.contollers.js

```diff
@@ -4,13 +4,22 @@ module.exports = {
   findAll: async (req, res) => {
     db.post
       .find(req.query)
+      .populate("user_id")
```

Comment:
I feel like you could use a virtual field to make this more readable on the client => it makes sense that you call this field `user_id` on the post since it simply references the user's id.  

The problem though is that when you then go populate, you break this "contract" and `user_id` is all of a sudden no longer and object id but a full user object.  You can get around this by using a virtual field called `user` that will get set on population only based on the local `user_id` field

check out the docs if you're curious
https://mongoosejs.com/docs/populate.html#populate-virtuals

and here are some other people talking about the same issue on github https://github.com/Automattic/mongoose/issues/1459